src/oversampling_methods/random_walk/interface_mlj.jl

```diff
@@ -30,6 +30,12 @@ Oversample data X, y using RandomWalkOversampler
 function MMI.transform(s::RandomWalkOversampler, _, X, y)
     random_walk_oversample(X, y; ratios = s.ratios, rng = s.rng, try_preserve_type=s.try_preserve_type)
 end
+function MMI.transform(s::RandomWalkOversampler, _, 
+                       X::AbstractMatrix{<:AbstractFloat}, 
+                       y::AbstractVector, 
+                       cat_inds::AbstractVector{<:Int})
```

Comment:
This is a problem. You can't pass "metadata" like `cat_inds` in this way. All the arguments after `_` must be "observational" data sharing a common number of observations (needed in algorithms that subsample). And currently you can't pass as kwarg either, because kwargs are not passed on  by `(::Machine, ...)`, for example. The only way would be to introduce a hyperparameter (field of the model struct). 

Maybe you just want to drop support for matrices in the MLJ interface.

Reply:
I was going to ask before doing this but when the tests I made for it passed I thought it should be perfectly okay. Alright will just revert back to the case where all methods have matrix support except purely heterogenous ones that apply logic differently to each kind (`SMOTENC` and `RandomWalkOversampler`).

Thank you for your comment.